apps/docs/src/app/examples/switches/SwitchWithIcon.tsx

```diff
@@ -0,0 +1,37 @@
+import { useState } from "react";
+import {
+  HvBaseSwitch,
+  HvButtonBase,
+  theme,
+} from "@hitachivantara/uikit-react-core";
+
+export default function Demo() {
+  const [checked, setChecked] = useState(false);
+
+  return (
+    <HvButtonBase
+      className="relative items-center rounded-full [&_.HvBaseSwitch-root]:bg-transparent"
+      onClick={() => setChecked((prev) => !prev)}
+    >
+      <HvBaseSwitch
+        size="medium"
+        value="on"
+        checked={checked}
+        aria-label="Switch"
+        onChange={(_evt, newChecked) => setChecked(newChecked)}
+        color={checked ? "warning" : "border"}
+      />
+      <div
+        className="bg-white border-1 border-borderString rounded-full w-28px h-28px flex items-center justify-center absolute top-2px cursor-pointer transition-left duration-200 ease"
```

Comment:
just an heads-up (not a request for changes), but I'm guessing that these absolute styles won't behave well when the Switch styles changes